icon_validator/rules/workflow_validators/workflow_screenshot_validator.py

```diff
@@ -99,38 +124,59 @@ def validate_screenshot_files_exist(self, spec):
                 self._files_list.append(file_name)
         except FileNotFoundError:
-            raise ValidationException(f"The screenshots directory could not be found at: {directory}\n"
-                                      "Please ensure that the screenshots directory exists.")
+            raise ValidationException(
+                f"The screenshots directory could not be found at: {directory}\n"
+                "Please ensure that the screenshots directory exists."
+            )
         if len(self._files_list) == 0:
-            raise ValidationException("There are no files in the screenshots directory."
-                                      " Please add at lest one screenshot.")
+            raise ValidationException(
+                "There are no files in the screenshots directory."
+                " Please add at lest one screenshot."
+            )
         for screenshot in self._files_list:
             if not screenshot.endswith(".png"):
-                raise ValidationException(f"All screenshots must be .png files. {screenshot} is not a .png file")
+                raise ValidationException(
+                    f"All screenshots must be .png files. {screenshot} is not a .png file"
+                )
 
     def validate_names_not_null(self):
         """
         Check that name keys in yaml are not null or blank strings.
         """
         for name in self._names_list:
             if not isinstance(name, str):
-                raise ValidationException("The name key for a screenshot must not be null")
+                raise ValidationException(
+                    "The name key for a screenshot must not be null"
+                )
         if any(names == "" for names in self._names_list):
-            raise ValidationException("The name key for a screenshot may not be a blank string")
+            raise ValidationException(
+                "The name key for a screenshot may not be a blank string"
+            )
 
-    def validate_screenshot_files_and_keys_match(self):
+    def validate_screenshot_files_and_keys_match(self, spec):
         """
         Check that screenshot file names match name keys in yaml.
         Check that there are not file names that do not exist in yaml and vice versa.
         """
-        sorted_names = sorted(self._names_list)
-        sorted_files = sorted(self._files_list)
-        if not sorted_files == sorted_names:
-            raise ValidationException("The screenshot files names and the screenshot names in the yaml do not match.")
+        d = spec.directory
+        screenshots_directory = f"{d}/screenshots"
+
+        screenshot_files = set(os.listdir(screenshots_directory))
+        spec_screenshots = {
+            name["name"] for name in spec.spec_dictionary()["resources"]["screenshots"]
+        }
+
+        mismatches = screenshot_files.difference(spec_screenshots)
+
+        if mismatches:
+            raise ValidationException(
```

Comment:
Can you print the mismatch here or give more info about what failed?

Reply:
Yep, that's an easy change.

Reply:
Now it says: 
```
[*] Executing validator WorkflowScreenshotValidator
Validator "WorkflowScreenshotValidator" failed! 
	Cause: Mismatch between provided screenshot files and screenshots in workflow.spec.yaml! The following mismatches were found: artifact1.png, job1.png
```